File: docs/snippets/Datamaps/datamap/xlsx-roundtrip.py
# docs:begin
from arctrl import DataContext, Datamap, XlsxController

leaf_area = DataContext(
    name="assays/Measurement/dataset/results.csv",
    label="leaf area",
)

datamap = Datamap([leaf_area])

workbook = XlsxController.Datamap().to_fs_workbook(datamap)

datamap_again = XlsxController.Datamap().from_fs_workbook(workbook)
# docs:end

# docs:assert
if len(datamap_again.DataContexts) != 1:
    raise Exception(
        "Expected one data context after xlsx workbook roundtrip, "
        f"got {len(datamap_again.DataContexts)}"
    )
# docs:endassert

File: docs/snippets/Datamaps/datamap/xlsx-roundtrip.ts
// docs:begin
import { DataContext, Datamap, XlsxController } from "@nfdi4plants/arctrl";

const leafArea = new DataContext(
  undefined,
  "assays/Measurement/dataset/results.csv",
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
  undefined,
  "leaf area",
);

const datamap = new Datamap([leafArea]);

const workbook = XlsxController.Datamap.toFsWorkbook(datamap);

const datamapAgain = XlsxController.Datamap.fromFsWorkbook(workbook);
// docs:end

// docs:assert
if (datamapAgain.DataContexts.length !== 1) {
  throw new Error(
    `Expected one data context after xlsx workbook roundtrip, got ${datamapAgain.DataContexts.length}`,
  );
}
// docs:endassert
